### PR TITLE
Ensure passwords are reloaded when updated via CLI

### LINKED ISSUE #598: lounge reset - reset password command requires a server restart.
In testing the tools for lounge.  The 'lounge reset' command does change the hash in the json file for the user but the password change doesn't take effect until the server is stopped and started again.  I've noticed in the clientManager.js that it checks for users being added or removed but from what I can tell it doesn't check for changes in the json file.

### PR DESCRIPTION
**Pull request by RJacksonm1**
As title, fixes #598

I didn't update the entire `config` object, cause I haven't the slightest clue what possible consequences that could have.

### CODE REVIEW

**Comment by xPaw** [comment]
This generally sounds fine, but this will start triggering `loadUser` every time Lounge itself updates the user file, which might be undesired.

**Comment by astorije** [comment]
Any alternative you can think of?

**Comment by RJacksonm1** [comment]
@xPaw Is the user config modified enough for that to be a performance concern?

File: src/clientManager.js [hunk 1]
```diff
@@ -78,12 +78,25 @@ ClientManager.prototype.loadUser = function(name) {
...
+		this.clients.push(client);
+	}
+
+	/**
+	 * If we happen to reload an existing client, make super duper sure we have their latest password.
+	 * We're not replacing the entire config object, because that could have undesired consequences.
+	 * 
+	 * @see https://github.com/thelounge/lounge/issues/598
+	 */
+	if (json.password !== undefined) {
```
Comment (xPaw) [inline src/clientManager.js]: Make this an `else if`

Comment (MaxLeiter) [reply src/clientManager.js]: bump, @RJacksonm1?

Comment (xPaw) [reply src/clientManager.js]: I'd wait for #1636.

**Comment by astorije** [comment]
@RJacksonm1, could you rebase this on master and fix conflicts please? Let us know if you get stuck

**Comment by astorije** [comment]
Hey @RJacksonm1, would you mind rebasing on `master` and fixing the conflicts, please? 🙏 
Would love to get your PR in! :)

**Comment by astorije** [comment]
Alright, I've rebased/fixed conflicts for this.

@xPaw, logs show `Password for user xxx was reset.` many times now, as user config gets reloaded over and over. Should we add something like `&& client.config.password !== userConfig.password` in that `if`? Do you think there is a better solution?

**Comment by xPaw** [comment]
@astorije Well obviously you want to only print that message if the password actually changed. However "shows many times" sounds like a bug, as I mentioned in first comment, lounge writing the file will trigger autoload.

**Comment by astorije** [comment]
> Well obviously you want to only print that message if the password actually changed. However "shows many times" sounds like a bug

Yes, I agree that we'll need to print the message if the password actually changed, but I left it that way for now specifically because of your second part: it does sound like a bug. Do you mind testing this locally and see if you see something obvious?

**Comment by astorije** [comment]
@xPaw, this should be ready now.

**Comment by astorije** [comment]
Hey @RJacksonm1, we have sticker packs for our contributors now!
If you're interested, go fill the form at https://goo.gl/forms/f5usqAEp5DWoeXC92 🙂

**Pull request merged**